client/state/happychat/middleware.js

```diff
@@ -51,9 +58,13 @@ export const socketMiddleware = ( connection = null ) => {
 
 			case HAPPYCHAT_BLUR:
 			case HAPPYCHAT_FOCUS:
-				store.dispatch( sendEvent( eventMessage[ action.type ] ) );
+				const state = store.getState();
+				isHappychatClientConnected( state ) && isHappychatChatAssigned( state )
+					? store.dispatch( sendEvent( eventMessage[ action.type ] ) )
```

Comment:
What would happen if `eventMessage[ action.type ]` is `undefined`? I think operators would receive an event message saying "undefined". Should we consider to default it to something else? 

I think it's going to be confusing for operators either way, so I'm fine with leaving it. Just wanted to raise the question :)

Reply:
fixed at b1f6938843a08bb7eeb7db625cab81135b0bd59e